pkg/ocm/billing.go

```diff
@@ -34,22 +36,61 @@ func (c *Client) GetBillingAccounts() ([]string, error) {
 		return nil, handleErr(quotaCostResponse.Error(), err)
 	}
 
-	var billingAccounts []string
+	var billingAccounts []*v1.CloudAccount
 	for _, item := range quotaCostResponse.Items().Slice() {
-		for _, account := range item.CloudAccounts() {
-			if account.CloudProviderID() == "aws" && !helper.Contains(billingAccounts, account.CloudAccountID()) {
-				billingAccounts = append(billingAccounts, account.CloudAccountID())
-			}
-		}
+		billingAccounts = append(billingAccounts, item.CloudAccounts()...)
 	}
 
 	if len(billingAccounts) == 0 {
-		return billingAccounts, errors.New("no billing accounts found")
+		return billingAccounts, errors.New("No billing account associated. " +
+			"Go to https://docs.openshift.com/rosa/rosa_architecture/rosa-understanding.html" +
+			" to learn how to associate your billing account." +
+			" You must associate a billing account to continue.")
 	}
 
 	return billingAccounts, nil
 }
 
+func GenerateBillingAccountsList(cloudAccounts []*v1.CloudAccount) []string {
+	var billingAccounts []string
+	for _, account := range cloudAccounts {
+		if account.CloudProviderID() == "aws" && !helper.ContainsPrefix(billingAccounts, account.CloudAccountID()) {
+			contracts, hasContract := account.GetContracts()
+			var contractString string
+			if hasContract {
+				dimensions := contracts[0].Dimensions()
+				for _, dimension := range dimensions {
+					if HasValidContracts(dimension) {
+						contractString = " [Contract enabled]"
+						break
+					}
+				}
+			}
+			billingAccounts = append(billingAccounts, account.CloudAccountID()+contractString)
+		}
+	}
+	return billingAccounts
+
+}
+
+func GetDimensionValue(dimension *v1.ContractDimension, resource string) int {
+	var value = 0
+	if dimension.Name() == resource {
+		value, _ = strconv.Atoi(dimension.Value())
+	}
+	return value
+}
+
+func HasValidContracts(dimension *v1.ContractDimension) bool {
+	if GetDimensionValue(dimension, "four_vcpu_hour") > 0 ||
+		GetDimensionValue(dimension, "control_plane") > 0 {
+		return true
+	}
+
+	return false
+
+}
```

Comment:
A ` HasValidContracts` function that accepts a cloudAccount parameter and iterates over its dimensions would make more sense to me.

Reply:
Refactored as: https://github.com/openshift/rosa/pull/1500/files#diff-54e259e8c0b289af4e61624a5e697f0285de9d9e0e14375c7d4fa82c772038eaR82